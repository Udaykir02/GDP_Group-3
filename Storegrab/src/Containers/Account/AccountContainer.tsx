import React, { useContext, useEffect } from 'react';
import { ScrollView, TouchableOpacity, View, StyleSheet } from 'react-native';
import { Text, Button, Switch } from 'react-native-paper'; // Import Button from react-native-paper

import { useAppTheme } from '../../styles/theme/theme';
import AppPageWrapper from '../../shared/AppPageWrapper';
import { useDispatch, useSelector } from 'react-redux'; // Import useDispatch
import { logoutRequest, resetRequest } from '../../../actions/userActions'; // Import your logout action creator
import { useNavigation } from '@react-navigation/native';
import MyIcon from '../../Components/MyIcon';
import { updateVendorAdmin } from '../../../reducers/users/slice';
import { resetCart } from '../../../reducers/cartReducer';
import { clearOrders } from '../../../reducers/orderReducer';
import { clearVendor } from '../../../reducers/vendorReducer';
import { clearState } from '../../../reducers/searchReducer';
import { clearLocation } from '../../../reducers/locationReducer';


const AccountContainer: React.FC = () => {
  const [isSwitchOn, setIsSwitchOn] = React.useState(false);

  const { user, vendorAdmin } = useSelector((state: any) => state.auth)

  // return <Switch value={isSwitchOn} onValueChange={onToggleSwitch} />;

  const { colors } = useAppTheme();
  const dispatch = useDispatch(); // Get dispatch function from react-redux
  const navigation = useNavigation()
  // Logout function
  const handleLogout = () => {
    dispatch(logoutRequest()); // Dispatch the logout action
    // Optionally, you can navigate to the login screen or perform any other action
  };

  // const handleNavigateToSubscription = () => {
  //   navigation.navigate('VendorSubscription');
  // }

  const handleSubscriptions = () => {
    navigation.navigate('VendorSubscription')
  }

  const onToggleSwitch = () => {
    dispatch(updateVendorAdmin(!vendorAdmin))
    dispatch(resetCart());
    dispatch(clearOrders());
    dispatch(clearVendor());
    dispatch(clearState());
    dispatch(clearLocation());
  };

  return (
    <AppPageWrapper>
      <ScrollView contentContainerStyle={styles.container}>
        <View style={styles.header}>
          {/* <Image
          source={{ uri: 'https://via.placeholder.com/150' }}
          style={styles.profileImage}
        /> */}
          <Text style={styles.name}>{user?.fname + " " + user?.lname}</Text>
          <Text style={styles.username}>@{user?.userId}</Text>
        </View>
        <View style={styles.details}>
          <Text style={styles.sectionTitle}>Contact Information</Text>
          <View style={styles.contactInfo}>
            <MyIcon name='Entypo|email' />
            <Text style={styles.contactText}>{user?.username}</Text>
          </View>
          <View style={styles.contactInfo}>

            <MyIcon name='Entypo|phone' />
            <Text style={styles.contactText}>not provided</Text>
          </View>
          <View style={styles.contactInfo}>

            <MyIcon name='Entypo|location-pin' />

            <Text style={styles.contactText}>San Francisco, CA</Text>
          </View>

        </View>
        <View style={styles.buttons}>
          <TouchableOpacity style={styles.button} onPress={() => { navigation.navigate('EditProfile') }}>
            <Text style={styles.buttonText}>Edit</Text>
          </TouchableOpacity>
          <TouchableOpacity style={styles.button} onPress={handleSubscriptions}>
            <Text style={styles.buttonText}>Subscriptions</Text>
          </TouchableOpacity>

        </View>
        <View style={[styles.contactInfo, { justifyContent: 'space-around', alignItems: 'center', alignContent: 'center' }]}>
          <Text style={[styles.sectionTitle, { alignSelf: 'center', marginTop: 7 }]}>Vendor Admin: </Text>
          <Switch value={vendorAdmin} onValueChange={onToggleSwitch} />
        </View>

        <TouchableOpacity style={styles.button} onPress={handleLogout}>
          <Text style={styles.buttonText}>Logout</Text>
        </TouchableOpacity>

      </ScrollView>
    </AppPageWrapper>
  );
};

const styles = StyleSheet.create({
  container: {
    flexGrow: 1,
    backgroundColor: '#f4f4f4',
    padding: 20,
  },
  header: {
    alignItems: 'center',
    marginBottom: 20,
  },
  profileImage: {
    width: 150,
    height: 150,
    borderRadius: 75,
    marginBottom: 10,
  },
  name: {
    fontSize: 24,
    fontWeight: 'bold',
  },
  username: {
    fontSize: 18,
    color: '#888',
  },
  details: {
    backgroundColor: '#fff',
    padding: 20,
    borderRadius: 10,
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.25,
    shadowRadius: 3.84,
    elevation: 5,
  },
  sectionTitle: {
    fontSize: 18,
    fontWeight: 'bold',
    marginBottom: 10,
  },
  sectionContent: {
    fontSize: 16,
    marginBottom: 20,
  },
  contactInfo: {
    flexDirection: 'row',
    alignItems: 'center',
    marginBottom: 10,
  },
  contactText: {
    fontSize: 16,
    marginLeft: 10,
  },
  socialLinks: {
    flexDirection: 'row',
    alignItems: 'center',
    marginBottom: 10,
  },
  socialText: {
    fontSize: 16,
    marginLeft: 10,
  },
  buttons: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    marginVertical: 20
  },
  button: {
    backgroundColor: '#4CAF50',
    padding: 10,
    borderRadius: 5,
    alignItems: 'center',
    flex: 1,
    marginHorizontal: 5,
  },
  buttonText: {
    color: '#fff',
    fontSize: 16,
  },
});

export default AccountContainer;
