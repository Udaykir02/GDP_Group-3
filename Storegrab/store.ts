import { configureStore } from '@reduxjs/toolkit';
import createSagaMiddleware from 'redux-saga';
import rootReducers from './reducers'; // Assuming rootReducers is defined in a separate file
import rootSaga from './sagas/rootSaga'; // Assuming rootSaga is defined in a separate file

// Create saga middleware instance
const sagaMiddleware = createSagaMiddleware();

// Create the Redux store with middleware and reducers
const store = configureStore({
  reducer: rootReducers,
  middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat(sagaMiddleware),
});

// Run the root saga
sagaMiddleware.run(rootSaga);

export default store;
