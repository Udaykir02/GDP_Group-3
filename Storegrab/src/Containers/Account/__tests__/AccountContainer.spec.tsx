import { Provider } from 'react-redux'
import store from '../../../../Store'
import render, {
    RenderResult,
} from '@testing-library/react-native/build/render'
import AccountContainer from '../AccountContainer'
import { NavigationContainer } from '@react-navigation/native'
jest.mock('@react-navigation/native', () => {
    return {
      ...jest.requireActual('@react-navigation/native'),
      useNavigation: () => ({
        navigate: jest.fn(),
      }),
    };
  });
const renderStartupContainerPage = () => {
    return render(
        <Provider store={store}>
            <NavigationContainer>
                <AccountContainer />
            </NavigationContainer>
        </Provider>,
    )
}
let page: RenderResult
describe('AccountContainer', () => {
    beforeEach(() => {
        jest.useFakeTimers();
        page = renderStartupContainerPage()
    })
    afterEach(() => {
        jest.runOnlyPendingTimers();
        jest.useRealTimers();
    });
    it('should render correctly', function () {
        expect(page.toJSON()).toMatchSnapshot()
    })
});