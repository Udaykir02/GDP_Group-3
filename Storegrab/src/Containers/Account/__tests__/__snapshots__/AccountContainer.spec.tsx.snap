// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountContainer should render correctly 1`] = `
<View
  style={
    {
      "backgroundColor": undefined,
      "flexDirection": "column",
      "padding": 16,
      "width": "100%",
    }
  }
>
  <RCTScrollView
    contentContainerStyle={
      {
        "backgroundColor": "#f4f4f4",
        "flexGrow": 1,
        "padding": 20,
      }
    }
  >
    <View>
      <View
        style={
          {
            "alignItems": "center",
            "marginBottom": 20,
          }
        }
      >
        <Text
          style={
            [
              {
                "textAlign": "left",
              },
              {
                "color": "rgba(28, 27, 31, 1)",
                "fontFamily": "System",
                "fontWeight": "400",
                "letterSpacing": 0,
              },
              {
                "writingDirection": "ltr",
              },
              {
                "fontSize": 24,
                "fontWeight": "bold",
              },
            ]
          }
        >
          undefined undefined
        </Text>
        <Text
          style={
            [
              {
                "textAlign": "left",
              },
              {
                "color": "rgba(28, 27, 31, 1)",
                "fontFamily": "System",
                "fontWeight": "400",
                "letterSpacing": 0,
              },
              {
                "writingDirection": "ltr",
              },
              {
                "color": "#888",
                "fontSize": 18,
              },
            ]
          }
        >
          @
        </Text>
      </View>
      <View
        style={
          {
            "backgroundColor": "#fff",
            "borderRadius": 10,
            "elevation": 5,
            "padding": 20,
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.25,
            "shadowRadius": 3.84,
          }
        }
      >
        <Text
          style={
            [
              {
                "textAlign": "left",
              },
              {
                "color": "rgba(28, 27, 31, 1)",
                "fontFamily": "System",
                "fontWeight": "400",
                "letterSpacing": 0,
              },
              {
                "writingDirection": "ltr",
              },
              {
                "fontSize": 18,
                "fontWeight": "bold",
                "marginBottom": 10,
              },
            ]
          }
        >
          Contact Information
        </Text>
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
              "marginBottom": 10,
            }
          }
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              [
                {
                  "color": undefined,
                  "fontSize": 12,
                },
                undefined,
                {
                  "fontFamily": "Entypo",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {},
              ]
            }
          >
            
          </Text>
          <Text
            style={
              [
                {
                  "textAlign": "left",
                },
                {
                  "color": "rgba(28, 27, 31, 1)",
                  "fontFamily": "System",
                  "fontWeight": "400",
                  "letterSpacing": 0,
                },
                {
                  "writingDirection": "ltr",
                },
                {
                  "fontSize": 16,
                  "marginLeft": 10,
                },
              ]
            }
          />
        </View>
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
              "marginBottom": 10,
            }
          }
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              [
                {
                  "color": undefined,
                  "fontSize": 12,
                },
                undefined,
                {
                  "fontFamily": "Entypo",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {},
              ]
            }
          >
            
          </Text>
          <Text
            style={
              [
                {
                  "textAlign": "left",
                },
                {
                  "color": "rgba(28, 27, 31, 1)",
                  "fontFamily": "System",
                  "fontWeight": "400",
                  "letterSpacing": 0,
                },
                {
                  "writingDirection": "ltr",
                },
                {
                  "fontSize": 16,
                  "marginLeft": 10,
                },
              ]
            }
          >
            not provided
          </Text>
        </View>
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
              "marginBottom": 10,
            }
          }
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              [
                {
                  "color": undefined,
                  "fontSize": 12,
                },
                undefined,
                {
                  "fontFamily": "Entypo",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {},
              ]
            }
          >
            
          </Text>
          <Text
            style={
              [
                {
                  "textAlign": "left",
                },
                {
                  "color": "rgba(28, 27, 31, 1)",
                  "fontFamily": "System",
                  "fontWeight": "400",
                  "letterSpacing": 0,
                },
                {
                  "writingDirection": "ltr",
                },
                {
                  "fontSize": 16,
                  "marginLeft": 10,
                },
              ]
            }
          >
            San Francisco, CA
          </Text>
        </View>
      </View>
      <View
        style={
          {
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginVertical": 20,
          }
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#4CAF50",
              "borderRadius": 5,
              "flex": 1,
              "marginHorizontal": 5,
              "opacity": 1,
              "padding": 10,
            }
          }
        >
          <Text
            style={
              [
                {
                  "textAlign": "left",
                },
                {
                  "color": "rgba(28, 27, 31, 1)",
                  "fontFamily": "System",
                  "fontWeight": "400",
                  "letterSpacing": 0,
                },
                {
                  "writingDirection": "ltr",
                },
                {
                  "color": "#fff",
                  "fontSize": 16,
                },
              ]
            }
          >
            Edit
          </Text>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#4CAF50",
              "borderRadius": 5,
              "flex": 1,
              "marginHorizontal": 5,
              "opacity": 1,
              "padding": 10,
            }
          }
        >
          <Text
            style={
              [
                {
                  "textAlign": "left",
                },
                {
                  "color": "rgba(28, 27, 31, 1)",
                  "fontFamily": "System",
                  "fontWeight": "400",
                  "letterSpacing": 0,
                },
                {
                  "writingDirection": "ltr",
                },
                {
                  "color": "#fff",
                  "fontSize": 16,
                },
              ]
            }
          >
            Subscriptions
          </Text>
        </View>
      </View>
      <View
        style={
          [
            {
              "alignItems": "center",
              "flexDirection": "row",
              "marginBottom": 10,
            },
            {
              "alignContent": "center",
              "alignItems": "center",
              "justifyContent": "space-around",
            },
          ]
        }
      >
        <Text
          style={
            [
              {
                "textAlign": "left",
              },
              {
                "color": "rgba(28, 27, 31, 1)",
                "fontFamily": "System",
                "fontWeight": "400",
                "letterSpacing": 0,
              },
              {
                "writingDirection": "ltr",
              },
              [
                {
                  "fontSize": 18,
                  "fontWeight": "bold",
                  "marginBottom": 10,
                },
                {
                  "alignSelf": "center",
                  "marginTop": 7,
                },
              ],
            ]
          }
        >
          Vendor Admin: 
        </Text>
        <RCTSwitch
          accessibilityRole="switch"
          onChange={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          onTintColor="rgba(103, 80, 164, 1)"
          style={
            {
              "height": 31,
              "width": 51,
            }
          }
          tintColor="rgba(103, 80, 164, 1)"
          value={false}
        />
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#4CAF50",
            "borderRadius": 5,
            "flex": 1,
            "marginHorizontal": 5,
            "opacity": 1,
            "padding": 10,
          }
        }
      >
        <Text
          style={
            [
              {
                "textAlign": "left",
              },
              {
                "color": "rgba(28, 27, 31, 1)",
                "fontFamily": "System",
                "fontWeight": "400",
                "letterSpacing": 0,
              },
              {
                "writingDirection": "ltr",
              },
              {
                "color": "#fff",
                "fontSize": 16,
              },
            ]
          }
        >
          Logout
        </Text>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
